src/watchmaker/workers/salt.py

```diff
@@ -661,6 +667,49 @@ def _set_grain(self, grain, value):
         self.log.info('Setting grain `%s` ...', grain)
         super(SaltLinux, self)._set_grain(grain, value)
 
+    def _install_pip_modules(self):
+        if self.pip_install:
+            python = self.run_salt(['grains.get', 'pythonexecutable'])
+            py_exec = python['stdout'].decode().split('\n')[1].strip()
+            self.log.info('Salt Python interpreter: `%s`', py_exec)
+
+            try:
+                ensure = self.call_process([py_exec, '-m', 'ensurepip'])
+                self.log.info('Ensure pip: %s', ensure['stdout'])
+                ver = self.call_process([py_exec, '-m', 'pip', '--version'])
+                self.log.info('Pip version: %s', ver['stdout'])
+            except WatchmakerException:
+                self.log.info('Pip not installed for Salt interpreter!')
+                self.log.info('Attempting pip install...')
+                get_pip = [
+                    'curl',
+                    'https://bootstrap.pypa.io/get-pip.py',
+                    '-o',
+                    'get-pip.py'
+                ]
+                self.call_process(get_pip)
+                self.call_process([py_exec, 'get-pip.py'])
+
+            for module in self.pip_install:
```

Comment:
One the bits about installing pip itself are separate, both linux and windows ought to be able to use the same logic to install packages with pip, so these bits can move into the SaltWorker instead of the Linux/Windows workers

Reply:
Pip can install multiple modules just fine, so no need to loop like this

Reply:
Can't move to SaltWorker since Linux/Windows are distinct (e.g., Windows is `self.run_salt()` vs. Linux is `self.call_process()`, Windows is comma separated vs. Linux is space separated)

Reply:
No, can use the same `call_process()` option on windows. Using salt with it's `pip.install` function _does not_ actually install to salt's python runtime anyway, you have to also pass the python script location to get it installed in the right location.

Reply:
argh - didn't realize. better to know now than when trying to run the formula!

Reply:
this is fixed